### PR TITLE
Speed at target min speed

### PR DESCRIPTION
**Pull request by Stifael**
FlightTaskAuto:
- move speed_at_target computation to Autoline
- keep a minimum speed of 0.5 when approaching a waypoint to prevent the vehicle from getting stuck

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Thanks! Best coordinated also with @bresch and @RomanBapst.

File: src/lib/FlightTasks/tasks/Auto/FlightTaskAuto.cpp [hunk 1]
```diff
@@ -575,63 +507,4 @@ bool FlightTaskAuto::_compute_heading_from_2D_vector(float &heading, Vector2f v)
...
-			  / (2.0f * middle_cruise_speed - _mc_cruise_speed - min_cruise_speed);
-		float c = _mc_cruise_speed - a;
-		float b = (middle_cruise_speed - c) / a;
-		speed_close = a * powf(b, angle) + c;
-	}
-
-	// speed_close needs to be in between max and min
-	return math::constrain(speed_close, min_cruise_speed, _mc_cruise_speed);
-}
+}
```
Comment (MaEtUgR) [inline src/lib/FlightTasks/tasks/Auto/FlightTaskAuto.cpp]: why is my added endline gone again? 😄

Comment (Stifael) [reply src/lib/FlightTasks/tasks/Auto/FlightTaskAuto.cpp]: ups

**Comment by MaEtUgR** [comment]
@bresch Asked for more context about the exponential function:

AFAIK the exponential function is an experimentally tested heuristic to set a reasonable target speed with which to pass the waypoint based on the corner angle to fly there, originally it was just an incremental improvement to breaking down to zero for every waypoint even though it might be just a camera waypoint and the flight path goes on straight. That heuristic stayed in there because there is no better solution yet. The function was there before, the pr just separates it into the functions library file. There are multiple such refactors in the pr removing duplicated/non-modular code.

**Pull request merged**